common/protocol_buffers/pb.h

```diff
@@ -0,0 +1,926 @@
+/* Common parts of the nanopb library. Most of these are quite low-level
```

Comment:
lets do the following: 
1. move the addition of protobuf as a separate independent PR with following things in mind
    a. first commit should link to the original commit files with a relevant reference of the specific commit
    b. second commit will incorporate all our relevant changes needed for our specific usage
3. move the entire protobuf from ~`common/protocol_buffers/`~ to `common/libraries/nanopb`